modules/tools/src/main/java/org/apache/ignite/tools/checkstyle/OverrideAnnotationOnTheSameLineCheck.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.tools.checkstyle;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
+
+/**
+ * <p>
+ * Checks that {@code @Override} annotations are located on the same line with target method declarartions.
+ * </p>
+ */
+public class OverrideAnnotationOnTheSameLineCheck extends AbstractCheck {
+    /** Different line error message. */
+    public static final String DIFF_LINE_ERR_MSG =
+        "@Override annotation on a different line than the target method declaration!";
+
+    /** No method declaration error message. */
+    public static final String NO_METHOD_ERR_MSG =
+        "No method declaration atfer @Override annotation!";
+
+    /** */
+    public static final String OVERRIDE = "Override";
+
+    /** {@inheritDoc} */
+    @Override public int[] getDefaultTokens() {
+        return new int[]{
+            TokenTypes.CLASS_DEF,
+            TokenTypes.INTERFACE_DEF,
+            TokenTypes.ENUM_DEF,
+            TokenTypes.METHOD_DEF,
+            TokenTypes.CTOR_DEF,
+        };
+    }
+
+    /** {@inheritDoc} */
+    @Override public int[] getAcceptableTokens() {
+        return new int[]{
+            TokenTypes.CLASS_DEF,
+            TokenTypes.INTERFACE_DEF,
+            TokenTypes.ENUM_DEF,
+            TokenTypes.METHOD_DEF,
+            TokenTypes.ANNOTATION_DEF,
+        };
+    }
+
+    /** {@inheritDoc} */
+    @Override public int[] getRequiredTokens() {
+        return CommonUtil.EMPTY_INT_ARRAY;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void visitToken(DetailAST ast) {
+        DetailAST nodeWithAnnotations = ast;
+
+        if (ast.getType() == TokenTypes.TYPECAST) {
```

Comment:
Here and below: Please, remove curly brackets for one-line statement after if

Reply:
fixed